File: Chapter 10/C#/Section_10_3_Resilience_Decorators/Resilient/Core/Domain/OrderStatus.cs
namespace Chapter10.Resilient.Core.Domain;

/// <summary>
/// DESIGN NOTE:
/// Enums provide the type-safety required for a robust system. By defining
/// this in the Core, we ensure that every adapter (FlakyPayments, Stripe, or SQL)
/// must translate its proprietary codes into our system's language.
/// </summary>
public enum OrderStatus
{
    PendingPayment,
    Paid,
    Failed
}

File: Chapter 10/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter10/section_10_3_resilience_decorators/resilient/core/domain/OrderStatus.java
package com.grokkingsoftwarearchitecture.chapter10.section_10_3_resilience_decorators.resilient.core.domain;

/**
 * * DESIGN NOTE:
 * Strictly enforcing system states via Enums prevents "Magic String" contamination. 
 * By defining this inside 'core/domain', we ensure the entire system speaks the 
 * same language, regardless of which external cloud provider we use.
 */
public enum OrderStatus {
    PENDING_PAYMENT,
    PAID,
    FAILED
}